diff --git utils/bazel/llvm-project-overlay/mlir/BUILD.bazel utils/bazel/llvm-project-overlay/mlir/BUILD.bazel
index 8559cc12cbaf..9cb9bb4b0609 100644
--- utils/bazel/llvm-project-overlay/mlir/BUILD.bazel
+++ utils/bazel/llvm-project-overlay/mlir/BUILD.bazel
@@ -10109,6 +10109,7 @@ cc_library(

 cc_library(
     name = "_mlir_c_runner_utils",
+    alwayslink = True,
     srcs = [
         "lib/ExecutionEngine/CRunnerUtils.cpp",
         "lib/ExecutionEngine/SparseTensorRuntime.cpp",
@@ -10147,6 +10148,7 @@ cc_binary(

 cc_library(
     name = "_mlir_runner_utils",
+    alwayslink = True,
     srcs = ["lib/ExecutionEngine/RunnerUtils.cpp"],
     hdrs = ["include/mlir/ExecutionEngine/RunnerUtils.h"],
     includes = ["include"],
